ejb3/src/main/java/org/jboss/as/ejb3/security/RunAsPrincipalInterceptor.java

```diff
@@ -61,9 +79,15 @@ public Object processInvocation(final InterceptorContext context) throws Excepti
         final SecurityIdentity oldIncomingRunAsIdentity = ejbComponent.getIncomingRunAsIdentity();
         SecurityIdentity newIdentity;
         try {
-            // The run-as-principal operation should succeed if the current identity is authorized to
-            // run as a user with the given name or if the caller has sufficient permission
-            if (runAsPrincipal.equals(ANONYMOUS_PRINCIPAL)) {
+            boolean remotingContextIsSet = remotingContextIsSet();
+            boolean isRemoteStatelessSecurityDisabled = !(ejbComponent instanceof StatefulSessionComponent) && remotingContextIsSet && !securityRequired;
```

Comment:
make principal anonymous and incoming identity null if  isRemoteStatelessSecurityDisabled

Reply:
not sure if this is the best way to identify stateless bean

Reply:
Hello @Skyllarr, you can use a instanceof check against `org.jboss.as.ejb3.component.stateless.StatelessSessionComponent` to be sure if it's a stateless EJB. In its current form, the above code in the PR, even singleton session bean component will pass this check. I don't know if you want that to happen.
Having said that, I'm just curious - is there a reason why this (security) logic is only applicable for stateless beans?

Reply:
@jaikiran Hello, thanks for review! I changed it to StatelessSessionComponent. This way the identity will not be propagated if the bean is a stateless unsecured remote bean. I did not find any reference for this, but failing tests lead me this way and it also made sense to me that I should not propagate identity for such a bean.